### PR TITLE
Relay: adding new API version 2018-preview

### PR DESCRIPTION
**Pull request by v-Ajnava**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [X] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [X] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [X] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [X] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by KrisBash — COMMENTED** [review]
Has this RP undergone an ARM review? If not, please schedule a meeting with arm api review dl

**Comment by v-Ajnava** [comment]
@KrisBash this is existing RP and PR is for new API

File: specification/relay/resource-manager/Microsoft.Relay/preview/2018-01-01-preview/PrivateEndpointConnection-preview.json [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+            "$ref": "#/parameters/privateEndpointConnectionName"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
```
[the remark this answers is not in the document]
Comment (AliMohsen) [reply specification/relay/resource-manager/Microsoft.Relay/preview/2018-01-01-preview/PrivateEndpointConnection-preview.json:173]: In our case delete is actually async. So we will actually be returning 202 along with an Azure-AsyncOperation header. 

At the moment we are returning 204, but a code fix has been made to return 202. It's just not deployed yet.

File: specification/relay/resource-manager/Microsoft.Relay/preview/2018-01-01-preview/PrivateEndpointConnection-preview.json [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "required": [
+        "location"
+      ],
+      "description": "Definition of resource."
+    },
+    "Resource": {
```
[the remark this answers is not in the document]
Comment (v-Ajnava) [reply specification/relay/resource-manager/Microsoft.Relay/preview/2018-01-01-preview/PrivateEndpointConnection-preview.json]: resolved.

File: specification/relay/resource-manager/Microsoft.Relay/preview/2018-01-01-preview/NetworkRuleSets-preview.json [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+      "properties": {
+        "ipMask": {
+          "type": "string",
+          "description": "IP Mask"
+        },
+        "action": {
+          "type": "string",
+          "description": "The IP Filter Action",
+          "enum": [
+            "Allow"
```
[the remark this answers is not in the document]
Comment (v-Ajnava) [reply specification/relay/resource-manager/Microsoft.Relay/preview/2018-01-01-preview/NetworkRuleSets-preview.json:144]: yes, we only support Allow.

File: specification/relay/resource-manager/Microsoft.Relay/preview/2018-01-01-preview/Namespaces-preview.json [hunk 4]
```diff
@@ -0,0 +1,741 @@
...
+        "None",
+        "InvalidName",
+        "SubscriptionIsDisabled",
+        "NameInUse",
+        "NameInLockdown",
+        "TooManyNamespaceInCurrentSubscription"
+      ],
+      "x-ms-enum": {
+        "name": "UnavailableReason",
+        "modelAsString": false
```
[the remark this answers is not in the document]
Comment (v-Ajnava) [reply specification/relay/resource-manager/Microsoft.Relay/preview/2018-01-01-preview/Namespaces-preview.json]: changed to true

**Review by KrisBash — APPROVED** [review]
Signing off for ARM

**Comment by v-Ajnava** [comment]
@NullMDR can you please review the PR?

**Comment by NullMDR** [comment]
Please update readme.go.md if you want to generate sdk for go for this new preview API

**Review by PhoenixHe-NV — APPROVED** [review]

**Pull request merged**